### PR TITLE
cleanup: warnings, time calculations and dhcp fixes (bsc#1188019)

### PR DESCRIPTION
**Pull request by mtomaschewski**
- Cleanup timer/timeout and lifetime types and calculations with very high or
  infinite lease time, which caused high cpu on RaspberryPi 2 (bsc#1188019)
- dhcp6: fix incorrect retransmission time randomization (re-)calculations
  USGv6: DHCPv6_7_1_05c, DHCPv6_7_1_05c, DHCPv6_7_1_05c
- fix warnings from more recent compilers

### CODE REVIEW

File: src/util.c [hunk 1]
```diff
@@ -1445,16 +1445,16 @@ ni_bool_t
...
+	int ret;
 
 	if (!fmt)
 		return FALSE;
 
 	va_start(ap, fmt);
 	ret = vasprintf(&path, fmt, ap) > 0;
 	va_end(ap);
-	if (!ret)
+	if (ret < 0 || !path)
```
Comment (cfconrad) [inline src/util.c:1457]: maybe it's time to switch to `ni_string_printf()` instead...

File: src/dhcp6/fsm.c [hunk 2]
```diff
@@ -2641,8 +2641,9 @@ ni_dhcp6_fsm_ia_addr_delete(ni_netdev_t *ifp, ni_dhcp6_device_t *dev, const ni_a
 		}
 	}
 
-	if (duplicate)
+	if (duplicate) {
```
Comment (cfconrad) [inline src/dhcp6/fsm.c:2644]: If I not mistaken, our code style is to avoid brackets on single statements

File: src/util.c (hunk 1 above)
Comment (mtomaschewski) [reply src/util.c:1457]: This wouldn't work but require a `ni_string_vprintf` not `ni_string_printf`. The intention was to fix a warning/bug, not to refactor the functions.

File: src/dhcp6/fsm.c (hunk 2 above)
Comment (mtomaschewski) [reply src/dhcp6/fsm.c]: yes.

File: src/util.c (hunk 1 above)
Comment (cfconrad) [reply src/util.c:1457]: ack

**Review by cfconrad — APPROVED** [review]
LGTM, thx!

**Pull request merged**